source/gui/accPropServer.py

```diff
@@ -90,24 +103,58 @@ def _getPropValue(self, pIDString, dwIDStringLen, idProp):
 			to extract the HWND/idObject/idChild from the identity string.
 			Note that, while one IAccPropServer implementation can annotate
 			multiple accessible elements, it is still bound to one wx.Control.
-		@type pIDString: str
 		@param dwIDStringLen: Specifies the length of the identity string specified by the pIDString parameter.
-		@type dwIDStringLen: int
-		@param idProp: Specifies a GUID indicating the desired property.
-		@type idProp: One of the oleacc.PROPID_* GUIDS
-		@return A tuple of the out params for the `IAccPropServer::GetPropValue` method: `VARIANT* pvarValue` and `BOOL*
-		pfHasProp`. When the pfHasProp part is FALSE / self.DOES_NOT_HAVE_PROP, then the pvarValue part must be VT_EMPTY.
-		Consider using self.NO_RETURN_VALUE instead. Returning (VT_EMPTY, HAS_PROP) IS valid, meaning the property exists
-		but is empty.
+		@param idProp: Specifies a GUID indicating the desired property. One of the values from oleacc.PROPID_*
+		@return Use L{self._hasProp} to return correct values or return None if unable to supply the property.
 		"""
 		raise NotImplementedError
 
-	def GetPropValue(self, pIDString, dwIDStringLen, idProp):
+	def _hasProp(
+			self,
+			value: AcceptedGetPropTypes
+	) -> Optional[Tuple[BOOL, AcceptedGetPropTypes]]:
+		"""Constructs a tuple for the `IAccPropServer::GetPropValue` method, two elements:
+			1. `VARIANT pvarValue`
+			2. `BOOL pfHasProp` (either self.HAS_PROP or self.DOES_NOT_HAVE_PROP)"""
+		return value, self.HAS_PROP
+
+	def GetPropValue(
+			self, this, # unused "this" used to indicate to comTypes we want a low level implementation
+			pIDString: str,
+			dwIDStringLen: int,
+			idProp: GUID,
+			pvarValue: POINTER(VARIANT),
+			pfGotProp: POINTER(BOOL)
+	) -> int:
+		""" Exposed method to get a prop value.
+		see L{_getPropValue} for more details of args.
+		Uses a low-level approach, because comtypes tries to clear the VARIANT even though it is an out param.
+		When the pfHasProp part is FALSE / self.DOES_NOT_HAVE_PROP, then the pvarValue.vt part must be VT_EMPTY.
+		"""
+		# ensure exceptions don't leave this function. They will get get swallowed by the caller.
+		# instead catch and log exceptions.
 		try:
-			return self._getPropValue(pIDString, dwIDStringLen, idProp)
-		except Exception:
+			# Preset values for "no prop value", in case we return early.
+			pfGotProp.contents.value = self.DOES_NOT_HAVE_PROP
+			pvarValue.contents.vt = VT_EMPTY
```

Comment:
Here you're setting contents.vt and below you're setting contents.value. Why is that?

If you prefer subscripting with [0], I can certainly live with that. Just not sure what's the preferred way of doing this.

Reply:
`.vt` must be set, to ensure that comtypes does not attempt to clear an uninitialised `VARIANT` , or more accurately, it's safe when it does clear it. When you assign a value, comtypes automatically converts the value being assigned, but first it clears the `VARIANT`. Out params don't seem to have a guarantee of being initialised. So `.vt` which, I assume stands for variant type, could be anything. It's explicitly unsafe to clear an uninitialised variant, see [variantClear Docs](https://docs.microsoft.com/en-us/windows/win32/api/oleauto/nf-oleauto-variantclear) for the following snippet:

> Do not use VariantClear on unitialized variants; use VariantInit to initialize a new VARIANTARG or VARIANT.

Rather than setting `.vt` we could call `VariantInit`
I suspect this was causing the issue I was having when trying to override name a while back.

I don't really have a preference for getting to the pointer contents, I'll leave it as is.

Reply:
I'm leaning towards using VariantInit rather than setting vt, but the [server annotation sample](https://docs.microsoft.com/en-gb/windows/win32/winauto/server-annotation-sample) does it like this.

Reply:
[This stackoverflow answer](https://stackoverflow.com/a/23542782) convinces me to use it.

Reply:
Sounds reasonable.